Java/src/main/java/Airport.java

```diff
@@ -77,56 +81,51 @@ public List<MilitaryPlane> getBomberMilitaryPlanes() {
 
     }
 
-    public List<experimentalPlane> getExperimentalPlanes() {
-        List<experimentalPlane> experimentalPlanes = new ArrayList<>();
-        for (Plane plane : planes) {
-            if (plane instanceof experimentalPlane) {
-                experimentalPlanes.add((experimentalPlane) plane);
+    public List<ExperimentalPlane> getExperimentalPlanes() {
+        List<ExperimentalPlane> experimentalPlanes = new ArrayList<>();
+        for (AbstractPlane plane : planes) {
+            if (plane instanceof ExperimentalPlane) {
+                experimentalPlanes.add((ExperimentalPlane) plane);
             }
         }
         return experimentalPlanes;
     }
 
     public Airport sortByMaxDistance() {
-        Collections.sort(planes, new Comparator<Plane>() {
-            public int compare(Plane o1, Plane o2) {
-                return o1.Get_Max_Flight_Distance() - o2.Get_Max_Flight_Distance();
+        Collections.sort(planes, new Comparator<AbstractPlane>() {
+            public int compare(AbstractPlane o1, AbstractPlane o2) {
+                return o1.getMaxFlightDistance() - o2.getMaxFlightDistance();
             }
         });
         return this;
     }
 
-
-    /**
-     * Sorts by max speed
-     * @return Airport
-     */
     public Airport sortByMaxSpeed() {
-        Collections.sort(planes, new Comparator<Plane>() {
-            public int compare(Plane o1, Plane o2) {
+        Collections.sort(planes, new Comparator<AbstractPlane>() {
+            public int compare(AbstractPlane o1, AbstractPlane o2) {
                 return o1.getMS() - o2.getMS();
             }
         });
         return this;
     }
 
     public Airport sortByMaxLoadCapacity() {
-        Collections.sort(planes, new Comparator<Plane>() {
-            public int compare(Plane o1, Plane o2) {
-                return o1.getMinLoadCapacity() - o2.getMinLoadCapacity();
+        Collections.sort(planes, new Comparator<AbstractPlane>() {
+            public int compare(AbstractPlane o1, AbstractPlane o2) {
+                return o1.getMaxLoadCapacity() - o2.getMaxLoadCapacity();
             }
         });
         return this;
     }
 
-    public List<? extends Plane> getPlanes() {
+    public List<? extends AbstractPlane> getPlanes() {
         return planes;
```

Comment:
You don't have to return the object field, either make in unmodifieble or create a new collection from your object state and return this newly created collection